File: Dockerfile.py
# Dockerfile for the Python model router service.
FROM python:3.12-slim AS base

WORKDIR /app

# Install uv
COPY --from=ghcr.io/astral-sh/uv:latest /uv /usr/local/bin/uv

# Copy and install deps
COPY services/model-router/pyproject.toml services/model-router/uv.lock ./
RUN uv sync --frozen --no-dev

# Copy source
COPY services/model-router/src ./src
COPY shared/agent-protocol/python ./shared/agent-protocol/python

ENV PYTHONPATH=/app/src:/app/shared/agent-protocol/python/src

EXPOSE 8081

CMD ["uv", "run", "uvicorn", "cascade_model_router.app:app", "--host", "0.0.0.0", "--port", "8081"]

File: Dockerfile.ts
# Shared Dockerfile for Bun-based TS services.
# Usage: docker build -f Dockerfile.ts --build-arg SERVICE_DIR=services/control-plane .
FROM oven/bun:1.3.14-slim AS base

ARG SERVICE_DIR

WORKDIR /app

# Copy lockfile and install deps
COPY package.json bun.lock ./
COPY ${SERVICE_DIR}/package.json ./package.json.tmp
RUN bun install --frozen-lockfile --production

# Copy the service source
COPY ${SERVICE_DIR} ./service
COPY shared/ ./shared

# Set the service-specific env
ENV NODE_ENV=production

# The service's entrypoint
CMD ["bun", "service/src/index.ts"]
